hypervisor/arch/x86/configs/vm_config.c

```diff
@@ -25,13 +26,27 @@ uint8_t get_vm_severity(uint16_t vm_id)
 }
 
 /**
- * return true if the input uuid is configured in VM
+ * return true if the input vm-name is configured in VM
  *
  * @pre vmid < CONFIG_MAX_VM_NUM
  */
-bool vm_has_matched_uuid(uint16_t vmid, const uint8_t *uuid)
+bool vm_has_matched_name(uint16_t vmid, const char *name)
 {
 	struct acrn_vm_config *vm_config = get_vm_config(vmid);
 
-	return (uuid_is_equal(vm_config->uuid, uuid));
+	return (strncmp(vm_config->name, name, MAX_VM_NAME_LEN) == 0);
+}
+
+uint16_t get_unused_vmid(void)
+{
+	uint16_t vm_id;
+	struct acrn_vm_config *vm_config;
+
+	for (vm_id = 0; vm_id < CONFIG_MAX_VM_NUM; vm_id++) {
+		vm_config = get_vm_config(vm_id);
+		if (vm_config->name[0] == '\0') {
+			break;
+		}
+	}
+	return (vm_id < CONFIG_MAX_VM_NUM) ? (vm_id) : (ACRN_INVALID_VMID);
```

Comment:
I was surprised to see use of the for loop iterator (vm_id) outside of the for loop, but I guess that's OK?

Reply:
for this api, it will return valid vm id(0, CONFIG_MAX_VM_NUM-1) if there is unused, otherwise it will return ACRN_INVALID_VMID, that is what we expected.